crates/kornia-imgproc/src/features/fast.rs

```diff
@@ -133,6 +155,236 @@ fn is_fast_corner(
+        } else {
+            update_indices(&consecutive_brighter, &idx, &mut start_idx, &mut end_idx, &mut most_consecutive);
+            update_indices(&consecutive_darker, &idx, &mut start_idx, &mut end_idx, &mut most_consecutive);
+
+            consecutive_brighter = 0;
+            consecutive_darker = 0;
+        }
+
+        if consecutive_brighter >= arc_length || consecutive_darker >= arc_length {
+            is_corner = true;
+        }
+
+        if consecutive_brighter >= 16 {
+            update_indices(&consecutive_brighter, &idx, &mut start_idx, &mut end_idx, &mut most_consecutive);
+            break;
+        } else if consecutive_darker >= 16 {
+            update_indices(&consecutive_darker, &idx, &mut start_idx, &mut end_idx, &mut most_consecutive);
+            break;
+        }
+    }
+
+    if !is_corner {
+        return (false, 0); // Not a corner found
+    }
+
+    // Sum of absolute differences for the corner score. 
+    let mut score = 0i32;
+    for offset in start_idx..start_idx + most_consecutive as usize {
+        let curr_idx = offset.rem_euclid(16); 
+        score += (center_pixel.abs_diff(*pixels[curr_idx]) - threshold) as i32;
+    }
+
+    (true, score)
+}
+
+/// Fast feature detector with Non-Maximum Suppression (NMS)
+/// Goal is to replace the previous `fast_feature_detector` implementation.
+///
+/// # Arguments
+///
+/// * `src` - The source image as Gray8 image.
+/// * `threshold` - The threshold for the fast feature detector.
+/// * `arc_length` - The total number of consecutive pixels in the Bresenham circle that must be brighter or darker than the center pixel.
+/// * `nms` - A boolean flag to enable or disable Non-Maximum Suppression.
+/// # Returns
+///
+/// A vector containing the coordinates of the detected keypoints.
+pub fn fast_feature_nms_detector(
+    src: &Image<u8, 1>,
+    threshold: u8,
+    arc_length: u8,
+    nms: bool
+) -> Result<Vec<[i32; 2]>, ImageError> {
+    let (cols, rows) = (src.cols() as i32, src.rows() as i32);
+
+    // Precompute the offsets for the Bresenham circle
+    let offsets = [
+        -3 * cols,     // 1
+        -3 * cols + 1, // 2
+        -2 * cols + 2, // 3
+        -cols + 3,     // 4
+        3,             // 5
+        cols + 3,      // 6
+        2 * cols + 2,  // 7
+        3 * cols + 1,  // 8
+        3 * cols,      // 9
+        3 * cols - 1,  // 10
+        2 * cols - 2,  // 11
+        cols - 3,      // 12
+        -3,            // 13
+        -cols - 3,     // 14
+        -2 * cols - 2, // 15
+        -3 * cols - 1,     // 16
+    ];
+
+    // Process rows in parallel
+    let (kp1, scores): (Vec<[i32; 2]>, Vec<i32>) = (3..rows - 3)
+        .into_par_iter()
+        .flat_map(|y| {
+            let row_start_idx = y * cols;
+            let mut row_keypoints = Vec::new();
+            let mut kp_scores = Vec::new();
+
+            for x in 3..cols - 3 {
+                let (is_corner, score) = get_fast_corner_score(
```

Comment:
```suggestion
                If let Some(score) = get_fast_corner_score(
```
Better use an optional and why not appending directly the FeaturePoint ?